shell/packages/sandstorm-db/db.js

```diff
@@ -560,6 +566,52 @@ isSignedUpOrDemo = function () {
   return false;
 }
 
+var calculateReferralBonus = function(accountId, plan) {
+  // This function returns an object of the form:
+  //
+  // - {grains: 0, storage: 0}
+  //
+  // which are extra resources this account gets as part of participating in the referral
+  // program. (Storage is measured in bytes, as usual for plans.)
+
+
+  // Authorization note: Only call this if accountId is the current user!
+  var isPaid = (plan && plan !== "free");
+
+  successfulReferralsCount = countReferrals(accountId);
+  if (isPaid) {
+    var maxPaidStorageBonus = 30 * 1e9;
```

Comment:
In the future, I can imagine we might tweak the referral program. Having the code depend on data in the database is error-prone and I can't say I love that pattern.

For now, I'd call this referral program experimental. I am already imagining how we might change it:

- Instead of "invite someone to a grain" being the thing that triggers the start of a referral program, we could do "suggest an app to someone and have them start a private grain of it".

So I'd rather keep this in the code and not move this into the database, but if you feel it's essential, I can do that.